torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -719,6 +719,50 @@ std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<
   return grad_inputs;
 }
 
+std::vector<Tensor> block_diag_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, const std::vector<ScalarType> &dtypes) {
```

Comment:
I mostly got this logic from our existing implementation of `cat_backward` (which knows how to deal with complex)